tomcat/src/main/java/org/apache/coyote/http/Dispatcher.java

```diff
@@ -1,146 +1,53 @@
 package org.apache.coyote.http;
 
-import com.techcourse.db.InMemoryUserRepository;
-import com.techcourse.model.User;
-import org.apache.coyote.http.request.HttpMethod;
+import com.techcourse.controller.BasicController;
+import com.techcourse.controller.LoginController;
+import com.techcourse.controller.RegisterController;
+import com.techcourse.controller.StaticResourceController;
+import org.apache.coyote.controller.Controller;
 import org.apache.coyote.http.request.HttpRequest;
 import org.apache.coyote.http.request.Path;
 import org.apache.coyote.http.response.HttpResponse;
-import org.apache.coyote.http.response.HttpStatus;
-import org.apache.coyote.http.response.ResponseHeader;
-import org.apache.coyote.http.response.StatusLine;
-import org.apache.coyote.util.StringUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.io.File;
-import java.io.IOException;
-import java.net.URL;
-import java.nio.file.Files;
 import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
 
 public class Dispatcher {
 
     private static final Logger log = LoggerFactory.getLogger(Dispatcher.class);
 
-    private static final String STATIC = "static";
-    private static final String REDIRECT = "/index.html";
-    private static final String UNAUTHORIZED = "/401.html";
+    private static final Map<String, Controller> controllers = new ConcurrentHashMap<>();
 
-    private static final SessionManager sessionManager = SessionManager.getInstance();
+    public Dispatcher() {
+        controllers.put("/", new BasicController());
+        controllers.put("/login", new LoginController());
+        controllers.put("/register", new RegisterController());
+    }
 
     public String dispatch(HttpRequest request) {
-        Path path = request.getRequestLine().getPath();
-
         try {
-            if (path.getUri().equals("/")) {
-                return HttpResponse.basicResponse().toResponse();
-            }
-            if (path.getUri().equals("/login")) {
-                return getLoginResponse(request);
-            }
-            if (path.getUri().equals("/register")) {
-                return getRegisterResponse(request);
-            }
-            return gerResourceResponse(path);
+            Controller controller = getController(request.getPath());
+            log.info("controller: {}", controller.getClass());
+            return controller.service(request).toResponse();
```

Comment:
service의 반환값이 뭐길래 toResponse를 하는 걸까? 라는 의문이 들었습니다..ㅎㅎ

그래서 나눠보면 어떨까 했는데 이렇게 하니 response.toResponse라는 메서드명이 또 눈에 들어오더라고요. 이미 객체 이름이 response인데 그 객체가 toResponse라는 메서드 이름을 갖는 것이 헷갈릴 수 있다고 생각해서 더불어 이 메서드명도 함께 다시 고민해보면 어떨까요?
```suggestion
            HttpResponse response = controller.service(request);
            return response.toResponse();
```

Reply:
이 부분은 위에 구구 리뷰에서 service 메서드의 시그니처가 바뀌면서 해소된 것 같습니다!
`void service(HttpRequest request, HttpResponse response)`
로 반환형이 바뀌면서 service 가 response 를 반환하는 불일치가 사라졌습니다!